grails-app/services/com/unifina/service/BalanceService.groovy

```diff
@@ -0,0 +1,58 @@
+package com.unifina.service
+
+import com.unifina.api.ApiException
+import groovy.transform.CompileStatic
+import org.web3j.exceptions.MessageDecodingException
+import org.web3j.protocol.Web3j
+import org.web3j.protocol.core.DefaultBlockParameterName
+import org.web3j.protocol.core.methods.response.EthGetBalance
+import org.web3j.protocol.http.HttpService
+
+import javax.annotation.PostConstruct
+import java.util.concurrent.ExecutionException
+
+class BalanceService {
+	protected Web3Balance web3
+
+	@PostConstruct
+	void init() {
+		final HttpService httpService = new HttpService("https://rinkeby.infura.io") // TODO: Fix address
```

Comment:
infura is fine I guess, though I think we'd want some amount of redundancy...

another thing is maybe using websockets instead of http, but for querying balances... whatever is fine